cni/network/invoker_cns.go

```diff
@@ -128,6 +131,11 @@ func (invoker *CNSIPAMInvoker) Add(addConfig IPAMAddConfig) (IPAMAddResult, erro
 		},
 	}
 
+	// ip-masq-agent handles snat in v4 overlay mode so just return here
+	if invoker.executionMode == util.V4OnlyOverlay {
```

Comment:
Instead of early return, can you call the SetHostOption if executionMode != util.v4OnlyOverlay